cli/medperf/commands/benchmark/run.py

```diff
@@ -0,0 +1,129 @@
+
+    def __init__(self, benchmark_uid: str, data_uid: str, ignore_errors=False):
+        self.benchmark_uid = benchmark_uid
+        self.data_uid = data_uid
+        self.model_execution = BenchmarkExecution(
+            benchmark_uid, data_uid, None, ignore_errors=ignore_errors,
+        )
+        self.outstanding_models = []
+        self.summary = {}
+
+    def prepare(self):
+        self.model_execution.prepare()
+        self.model_execution.model_uid = self.model_execution.benchmark.models[0]
+
+    def validate(self):
+        self.model_execution.validate()
+
+    def prepare_cubes(self):
+        self.model_execution.evaluator = self.model_execution.__get_cube(
+            self.model_execution.benchmark.evaluator, "Evaluator"
+        )
+        self.__filter_models()
+
+    def __filter_models(self):
+        benchmark_models = self.model_execution.benchmark.models
+        results = Result.all()
+        results = [
+            result
+            for result in results
+            if result.benchmark_uid == self.benchmark_uid
+            and result.dataset_uid == self.data_uid
+        ]
+        done_models = [result.model_uid for result in results]
+        self.outstanding_models = [
+            model for model in benchmark_models if model not in done_models
+        ]
+        self.summary["skipped_models"] = len(done_models)
+
+    def run_models(self):
+        self.summary["executions"] = []
+        for model in self.outstanding_models:
+            exec_summary = {
+                "model": model,
+                "success": False,
+                "partial": "",
+                "error": "",
+            }
+            # reset
+            self.model_execution.model_uid = model
+            self.model_execution.out_path = results_path(
+                self.benchmark_uid, model, self.model_execution.dataset.uid
+            )
+            self.model_execution.metadata["partial"] = False
+
+            try:
+                self.model_execution.model_cube = self.model_execution.__get_cube(
+                    model, "Model"
+                )
+            except InvalidEntityError as e:
+                config.ui.print_error(
+                    f"There was an error when retrieving the model mlcube {model}: {e}"
+                )
+                exec_summary["error"] = str(e)
+                self.summary["executions"].append(exec_summary)
+                continue
+            try:
+                self.model_execution.run_cubes()
+            except ExecutionError as e:
+                config.ui.print_error(
+                    f"There was an error when executing the benchmark with the model {model}: {e}"
+                )
+                exec_summary["error"] = str(e)
+                self.summary["executions"].append(exec_summary)
+                continue
+
+            self.model_execution.write()
+            self.model_execution.remove_temp_results()
+
+            exec_summary["success"] = True
```

Comment:
I like the idea of handling errors and providing a summary of the complete execution here. Nice!